src/test/groovy/com/money_account_service/services/TopUpServiceContractTest.groovy

```diff
@@ -0,0 +1,89 @@
+package com.money_account_service.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.money_account_service.dtos.request.CreateAccountRequestDto
+import com.money_account_service.dtos.request.TopUpRequestDto
+import com.money_account_service.dtos.response.AuthorizeResponseDto
+import com.money_account_service.dtos.response.CreateAccountResponseDto
+import com.money_account_service.dtos.response.TopUpResponseDto
+import com.money_account_service.entities.AccountEntity
+import com.money_account_service.entities.TransferEntity
+import com.money_account_service.repositories.AccountRepository
+import com.money_account_service.repositories.TransferRepository
+import com.money_account_service.utility.UserServiceClient
+import org.spockframework.spring.SpringBean
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.HttpHeaders
+import org.springframework.test.context.TestPropertySource
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import spock.lang.Specification
+
+import java.time.Clock
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@TestPropertySource(locations = "classpath:application-test.properties")
+class TopUpServiceContractTest extends Specification {
+
+    @Autowired
+    private MockMvc mockMvc
+    @Autowired
+    private ObjectMapper objectMapper
+    @SpringBean
+    private final TransferRepository transferRepository = Mock(TransferRepository)
```

Comment:
I really don't like mocking the repository, instead of using dockerized (e.g. from TestContainers). This is a huge anti-pattern and very likely to be error-prone. 

You can have all the DB logic incorrect and those test will be green. E.g. you can have incorrect DB schema & Spring repository, but mocking will allow it to work:
```
transferRepository.save(_ as TransferEntity) >> transferEntity
```